civis/io/_tables.py

```diff
@@ -1040,7 +1052,7 @@ def civis_file_to_table(file_id, database, table, client=None,
         cleaning_futures, client, headers, need_table_columns, delimiter
     )
 
-    table_columns = table_columns or cleaned_table_columns
+    table_columns = cleaned_table_columns if need_table_columns else table_columns
```

Comment:
This logic was flawed, since we may need to clean even when table_columns are provided, so needed to be updated.